src/Avalonia.Base/Rendering/Composition/Server/ServerCompositionTarget.cs

```diff
@@ -35,10 +36,24 @@ internal partial class ServerCompositionTarget : IDisposable
         private HashSet<ServerCompositionVisual> _attachedVisuals = new();
         private Queue<ServerCompositionVisual> _adornerUpdateQueue = new();
 
+        public long Id { get; }
+        public ulong Revision { get; private set; }
         public ICompositionTargetDebugEvents? DebugEvents { get; set; }
         public ReadbackIndices Readback { get; } = new();
         public int RenderedVisuals { get; set; }
 
+        private DiagnosticTextRenderer DiagnosticTextRenderer
```

Comment:
`GlyphRun` instances need to be created on the UI thread, it can cause issues otherwise. We can probably use a single `DiagnosticTextRenderer` instance created with the `Compositor`

Reply:
I see. I wanted to avoid allocating the glyph runs completely if there's no diagnostic. Changing that back.